src/support/cond_auto.c

```diff
@@ -62,8 +62,8 @@ __wt_cond_auto_wait_signal(WT_SESSION_IMPL *session, WT_CONDVAR *cond, bool prog
          * - it's not necessary for the previous wait time to be updated every time.
          */
         WT_ORDERED_READ(saved_prev_wait, cond->prev_wait);
-        __wt_atomic_cas64(
-          &cond->prev_wait, saved_prev_wait, WT_MIN(cond->max_wait, saved_prev_wait + delta));
+        WT_IGNORE_RET_BOOL(__wt_atomic_cas64(
```

Comment:
On thinking about this, I think it might be better to increment a statistic if this compare and swap fails. Could you make that change instead?

Reply:
Fixed as suggested.